frontend/src/components/resourceMap/sources/GraphSources.tsx

```diff
@@ -232,13 +232,17 @@ export function GraphSourceManager({ sources, children }: GraphSourceManagerProp
 
   const contextValue = useThrottledMemo(
     () => {
-      const nodes: GraphNode[] = [];
-      const edges: GraphEdge[] = [];
+      let nodes: GraphNode[] = [];
+      let edges: GraphEdge[] = [];
 
       selectedSources.forEach(id => {
         const data = sourceData.get(id);
-        nodes.push(...(data?.nodes ?? []));
-        edges.push(...(data?.edges ?? []));
+        if (data?.nodes) {
+          nodes = nodes.concat(data.nodes);
+        }
+        if (data?.edges) {
+          edges = edges.concat(data.edges);
+        }
```

Comment:
I think push is faster than concat, is there another reason to make this change?

Reply:
push(...arr) crashes if array is larger than the call stack can hold, and this fixes that. 
concat works but you're right it's not the fastest, although this is not in a hot path. I'll check other solutions

Reply:
I did some simple tests and it looks like concat is consistently faster, maybe this case is optimized by v8 or something. keeping it like this